### PR TITLE
[kube-prometheus-stack] Add docs on podmonitors

### PR DESCRIPTION
**Pull request by saada**
#### What this PR does / why we need it:
Add docs on podmonitors

#### Checklist
- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by saada** [comment]
Not sure I need to bump the version for a doc change? 

```
 ✖︎ kube-prometheus-stack => (version: "9.4.4", path: "charts/kube-prometheus-stack") > Chart version not ok. Needs a version bump!
```

File: charts/kube-prometheus-stack/README.md [hunk 1]
```diff
@@ -196,9 +196,12 @@ For more in-depth documentation of configuration options meanings, please see
 
 ## prometheus.io/scrape
 
-The prometheus operator does not support annotation-based discovery of services, using the `serviceMonitor` CRD in its place as it provides far more configuration options. For information on how to use servicemonitors, please see the documentation on the `prometheus-operator/prometheus-operator` documentation here: [Running Exporters](https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/user-guides/running-exporters.md)
+The prometheus operator does not support annotation-based discovery of services, using the `podMonitor` or `serviceMonitor` CRD in its place as they provide far more configuration options. For information on how to use podmonitors and servicemonitors, please see the documentation on the `prometheus-operator/prometheus-operator` documentation here:
```
Comment (scottrigby) [inline charts/kube-prometheus-stack/README.md:199]: Suggestions:

1. The CRDs should be CamelCased, right?
    - `podMonitor`-> `PodMonitor`
    - `serviceMonitor` -> `ServiceMonitor`
1. "podmonitors and servicemonitors" -> "PodMonitors/ServiceMonitors" for consistency with your other paragraph change below.
1. Added line break, for one sentence per line md style rule (not yet able to add to automated linting. See https://github.com/prometheus-community/helm-charts/issues/25)


```suggestion
The prometheus operator does not support annotation-based discovery of services, using the `PodMonitor` or `ServiceMonitor` CRD in its place as they provide far more configuration options.
For information on how to use podmonitors and servicemonitors, please see the documentation on the `prometheus-operator/prometheus-operator` documentation here:
```

File: charts/kube-prometheus-stack/README.md [hunk 2]
```diff
@@ -196,9 +196,12 @@ For more in-depth documentation of configuration options meanings, please see
...
+By default, Prometheus discovers PodMonitors and ServiceMonitors within its namespace, that are labeled with the same release tag as the prometheus-operator release. Sometimes, you may need to discover custom PodMonitors/ServiceMonitors, for example used to scrape data from third-party applications. An easy way of doing this, without compromising the default PodMonitors/ServiceMonitors discovery, is allowing Prometheus to discover all PodMonitors/ServiceMonitors within its namespace, without applying label filtering. To do so, you can set `prometheus.prometheusSpec.podMonitorSelectorNilUsesHelmValues` and `prometheus.prometheusSpec.serviceMonitorSelectorNilUsesHelmValues` to `false`.
```
Comment (scottrigby) [inline charts/kube-prometheus-stack/README.md:204]: Only added Line breaks for #25. See review comment above

```suggestion
By default, Prometheus discovers PodMonitors and ServiceMonitors within its namespace, that are labeled with the same release tag as the prometheus-operator release.
Sometimes, you may need to discover custom PodMonitors/ServiceMonitors, for example used to scrape data from third-party applications.
An easy way of doing this, without compromising the default PodMonitors/ServiceMonitors discovery, is allowing Prometheus to discover all PodMonitors/ServiceMonitors within its namespace, without applying label filtering.
To do so, you can set `prometheus.prometheusSpec.podMonitorSelectorNilUsesHelmValues` and `prometheus.prometheusSpec.serviceMonitorSelectorNilUsesHelmValues` to `false`.
```

**Review by scottrigby — CHANGES_REQUESTED** [review]
@saada thanks for your contribution! Nice find. Just a few comments

File: charts/kube-prometheus-stack/README.md (hunk 1 above)
Comment (scottrigby) [reply charts/kube-prometheus-stack/README.md:199]: Updated my commit suggestion

**Comment by scottrigby** [comment]
@saada Yes every chart change requires bumping the version, as chart packages must be immutable. See https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#versioning for details 😄

**Review by scottrigby — APPROVED** [review]
💯

**Pull request merged**